### PR TITLE
Fixes #3209 - fixes misplacement of dot in validation messages.

### LINKED ISSUE #3209: Validation point position for containers is placed wrong according to #3175
Validation for container objects yields a validation error message, which contains a misplaced point as shown here:
`"container.volumes.[0]hostPath"`
This issue was discovered in #3175

### PR DESCRIPTION
**Pull request by alexanderweber**
Fixes an issue of point misplacement as described in #3209

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/api/v2/Validation.scala [hunk 1]
```diff
@@ -52,22 +52,28 @@ object Validation {
...
       case r: RuleViolation => Set(parentDesc.map(p =>
-        r.withDescription(concatPath(p, r.description, attachDot)))
+        r.withDescription(concatPath(p, r.description, attachDot = false)))
         .getOrElse(r))
       case g: GroupViolation => g.children.flatMap { c =>
         parentDesc.map { p =>
-          val desc = concatPath(p, g.description, attachDot)
-          allRuleViolationsWithFullDescription(c, Some(desc), g.description.isDefined)
+          val forceNoDot = g.value match {
+            case x :: xs => false
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/api/v2/Validation.scala]: Here you pattern match for a non empty list. Perhaps you want to do something like: 
```scala
case _: Iterable => false
```

File: src/main/scala/mesosphere/marathon/api/v2/Validation.scala [hunk 2]
```diff
@@ -52,22 +52,28 @@ object Validation {
...
+            case _       => true
+          }
+          val desc = concatPath(p, g.description, forceNoDot)
+          allRuleViolationsWithFullDescription(c, Some(desc))
         } getOrElse {
-          allRuleViolationsWithFullDescription(c, g.description, g.description.isDefined)
+          val desc = if (g.description.isDefined)
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/api/v2/Validation.scala]: `g.description` is already n option. You could use it like:
```scala
g.description.map( concatPath("", _, attachDot = true))
```

**Comment by aquamatthias** [comment]
Thanks! 
Small improvement suggestions - otherwise fine.

File: src/main/scala/mesosphere/marathon/api/v2/Validation.scala (hunk 1 above)
Comment (alexanderweber) [inline src/main/scala/mesosphere/marathon/api/v2/Validation.scala]: You are right, I did change that to the suggested line.

File: src/main/scala/mesosphere/marathon/api/v2/Validation.scala (hunk 2 above)
Comment (alexanderweber) [inline src/main/scala/mesosphere/marathon/api/v2/Validation.scala]: Missed that one, thanks for pointing that out!

**Comment by alexanderweber** [comment]
@aquamatthias your suggestions made it into a new commit. Tests are good, looking forward to merging.

**Comment by aquamatthias** [comment]
Thanks! :+1:

**Pull request merged**